### PR TITLE
Bundle size reduction

### LINKED ISSUE #313: UTBotJava IntelliJ-plugin bundle is too heavy
**Description**

We have recently reduced the size of the plugin bundle. But it is still rather big.

**To Reproduce**

Open the `publish-plugin-and-cli.yml` action, check the bundle size.

**Expected behavior**

More reasonable plugin size.

**Actual behavior**

Bundle size is 302 MB

**Visual proofs (screenshots, logs, images)**

Capture

**Environment**

GitHub, Gradle

**Additional context**

No additional context.

### PR DESCRIPTION
**Pull request by denis-fokin**
# Description

I have removed some dependencies from Gradle files.

Fixes #313

## Type of Change

Please delete options that are not relevant.

- Bug fix (non-breaking change which fixes an issue)

# How Has This Been Tested?

## Manual Scenario 

# Checklist (remove irrelevant options):

- [x] The change followed the style guidelines of the UTBot project
- [x] Self-review of the code is passed
- [x] The change contains enough commentaries, particularly in hard-to-understand areas
- [ ] New documentation is provided or existed one is altered
- [x] No new warnings
- [ ] Tests that prove my change is effective
- [x] All tests pass locally with my changes

### CODE REVIEW

File: utbot-intellij/build.gradle [hunk 1]
```diff
@@ -21,20 +21,21 @@ buildscript {
 
 dependencies {
     api project(':utbot-summary')
```
Comment (IdeaSeeker) [inline utbot-intellij/build.gradle:23]: `utbot-summary` dependency can be removed (?)

File: utbot-cli/build.gradle [hunk 2]
```diff
@@ -25,6 +24,8 @@ dependencies {
     compile group: 'org.apache.logging.log4j', name: 'log4j-core', version: log4j2_version
     compile group: 'org.apache.logging.log4j', name: 'log4j-api', version: log4j2_version
     implementation group: 'org.jacoco', name: 'org.jacoco.report', version: jacoco_version
+
+    compile project(":utbot-analytics")
```
[the remark this answers is not in the document]
Comment (denis-fokin) [reply utbot-cli/build.gradle:28]: I asked QA to take a look at the build. If no additional problems found, l will remove the dependency

File: utbot-instrumentation/build.gradle [hunk 3]
```diff
@@ -27,7 +22,7 @@ jar {
                 'Premain-Class': 'org.utbot.instrumentation.agent.Agent',
         )
     }
-    from { configurations.compileClasspath.collect { it.isDirectory() ? it : zipTree(it) } }
+    from { configurations.runtimeClasspath.collect { it.isDirectory() ? it : zipTree(it) } }
```
[the remark this answers is not in the document]
Comment (denis-fokin) [reply utbot-instrumentation/build.gradle:30]: I saw this project, but have not tried yet

File: utbot-intellij/build.gradle (hunk 1 above)
Comment (denis-fokin) [reply utbot-intellij/build.gradle]: Summary is responsible for comments generation, so let's preserve them

File: utbot-cli/build.gradle (hunk 2 above)
[the remark this answers is not in the document]
Comment (denis-fokin) [reply utbot-cli/build.gradle]: `utbot-summary` is needed for comments generation

**Review by dtim — APPROVED** [review]

**Pull request merged**